### ISSUE TITLE
You can open multiple of the same viewer

### ISSUE THREAD

**Issue by briansrls** [visualboyadvance-m/visualboyadvance-m#734]
### Expected behavior
Attempting to open a viewer that is already opened should focus that viewer instead.

### Actual behavior
It opens another copy of it.

### Steps to reproduce the behavior
Open a viewer, say Disassemble viewer, and then try to open it again.

I'm not sure if it's a bug, but it doesn't seem like great UI design.  This is the only program in my memory that behaves this way, so likely a bug.

I was thinking that maybe it would be valid to have multiple Memory viewers open (for more screen real estate).  But that argument doesn't really work well for the other viewers.

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.):
Windows

If using Linux, specify if using xorg or Wayland:
xorg

Version of code (pre-built binary version, or commit ref, or just "master"):
master

**Comment by rkitover** [comment]
All the viewer stuff is horribly broken, we need a lot more work in this area.

**Comment by briansrls** [comment]
@rkitover maybe we could consolidated all UI changes into one giant issue/rewrite?  Would help us differentiate between issues/nice to haves, like this.

**Comment by rkitover** [comment]
I was planning at some point to go through all the issues and file everything appropriately, set up more project boards, etc.. Not a high a priority right now, last month or so I've been really busy keeping the CI up and running and fixing the fallout from various recent changes, like the joystick stuff.

We need to take care of the assertion failure you mentioned, that's from denisfa's background input work.

Highest priority right now is stabilizing everything and getting the release out.

**Comment by briansrls** [comment]
OK, want to make a list of musts for the release?

**Comment by rkitover** [comment]
Actually, I would be eternally grateful if you can look at some of the cheat codes issues, I'm fairly sure that's entirely a minor UI issue, such as #653.

**Comment by rkitover** [comment]
The core should handle all cheat codes stuff just fine, it's just all not connected properly.

**Comment by rkitover** [comment]
And can you look at #741 please when you get a chance?

**Comment by briansrls** [comment]
Yes, I will look at your PR when I can, I think I actually have a mayflash I can use.  And I will look at #653 next
